experiments/storm/src/main/scala/io/gearpump/experiments/storm/processor/StormProcessor.scala

```diff
@@ -19,96 +19,89 @@
-  private val boltId = conf.getString(GraphBuilder.COMPONENT_ID).getOrElse(
-    throw new RuntimeException(s"Storm bolt id not found for processor $pid"))
-  private val boltSpec = conf.getValue[Bolt](GraphBuilder.COMPONENT_SPEC).getOrElse(
-    throw new RuntimeException(s"Storm bolt spec not found for processor $pid"))
+
+  private val topology = conf.getValue[StormTopology](TOPOLOGY).get
+  private val boltSpec = conf.getValue[Bolt](COMPONENT_SPEC).get
   private val bolt = Utils.getSetComponentObject(boltSpec.get_bolt_object()).asInstanceOf[IBolt]
-  private val stormConfig = getStormConfig(conf, boltSpec.get_common())
-  private val topologyContextBuilder = TopologyContextBuilder(topology, stormConfig,
-    multiLang = bolt.isInstanceOf[ShellBolt])
-  private val outputCollector = new StormOutputCollector(taskContext, pid, boltId)
+  private val stormConfig = parseJsonStringToMap(boltSpec.get_common().get_json_conf())
+  private var taskToTopologyContext = Map.empty[Integer, TopologyContext]
+  private var topologyContextBuilder: TopologyContextBuilder = null
+  private val stormTaskId = gearpumpTaskIdToStorm(taskContext.taskId)
+  private var outputCollector: StormOutputCollector = null
   private var scheduler: Cancellable = null
 
 
   override def onStart(startTime: StartTime): Unit = {
-    Option(stormConfig.get(TICK_TUPLE_FREQ_SECS)).foreach { freq =>
-      val freq = stormConfig.get(TICK_TUPLE_FREQ_SECS).asInstanceOf[java.lang.Long]
-      val values = new JArrayList[Object](1)
-      values.add(freq)
-      val tickTuple = new TupleImpl(getTickTupleContext, values, SYSTEM_TASK_ID, SYSTEM_STREAM_ID, null)
-      scheduler = taskContext.schedule(new FiniteDuration(freq, TimeUnit.SECONDS),
-        new FiniteDuration(freq, TimeUnit.SECONDS))({
-        self ! tickTuple
-      })
-    }
+    implicit val timeout = Constants.FUTURE_TIMEOUT
+    val dagFuture = askAppMasterForDAG(taskContext.appMaster)
+    val dag = Await.result(dagFuture, timeout.duration)
+
+    val taskToComponent = getTaskToComponent(dag)
+    topologyContextBuilder = new TopologyContextBuilder(topology, stormConfig, taskToComponent)
+    val topologyContext = topologyContextBuilder.build(stormTaskId)
+
+    taskToTopologyContext += stormTaskId -> topologyContext
 
+    outputCollector = new StormOutputCollector(taskContext, topologyContext)
     val delegate = new StormBoltOutputCollector(outputCollector)
-    val topologyContext = topologyContextBuilder.buildContext(pid, boltId)
     bolt.prepare(stormConfig, topologyContext, new OutputCollector(delegate))
+
+    setupTickTuple(topologyContext)
   }
 
   override def onNext(msg: Message): Unit = {
     val stormTuple = msg.msg.asInstanceOf[StormTuple]
     outputCollector.setTimestamp(msg.timestamp)
-    bolt.execute(stormTuple.toTuple(topologyContextBuilder))
+    val sourceId = stormTuple.stormTaskId
+    val topologyContext = taskToTopologyContext.getOrElse(sourceId, topologyContextBuilder.build(sourceId))
```

Comment:
Why will you create different topologyContext? Is this different from the toplogyContext in onStart?